core/src/main/scala/org/bitcoins/core/psbt/PSBTHelper.scala

```diff
@@ -0,0 +1,225 @@
+package org.bitcoins.core.psbt
+
+import org.bitcoins.core.crypto.{ECDigitalSignature, ECPublicKey, ExtPublicKey}
+import org.bitcoins.core.hd.BIP32Path
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.CompactSizeUInt
+import org.bitcoins.core.protocol.script.{
+  ScriptPubKey,
+  ScriptSignature,
+  WitnessScriptPubKey
+}
+import org.bitcoins.core.protocol.transaction.{Transaction, WitnessTransaction}
+import org.bitcoins.core.psbt.GlobalPSBTRecord._
+import org.bitcoins.core.psbt.InputPSBTRecord._
+import org.bitcoins.core.psbt.PSBTGlobalKey._
+import org.bitcoins.core.psbt.PSBTInputKey._
+import org.bitcoins.core.script.crypto.HashType
+import org.bitcoins.core.serializers.script.RawScriptWitnessParser
+import org.bitcoins.core.util.BitcoinScriptUtil
+import scodec.bits.ByteVector
+
+import scala.annotation.tailrec
+
+object PSBTHelper {
+
+  private def parseKeyAndValue(
+      bytes: ByteVector): (Long, ByteVector, ByteVector) = {
+    val keySize = CompactSizeUInt.parse(bytes)
+    val key = bytes.drop(keySize.size).take(keySize.toLong)
+    val valueSize =
+      CompactSizeUInt.parse(bytes.drop(keySize.size + keySize.toLong))
+    val value = bytes
+      .drop(keySize.size + keySize.toLong + valueSize.size)
+      .take(valueSize.toLong)
+    val totalSize = keySize.size + keySize.toLong + valueSize.size + valueSize
+      .toLong
+    (totalSize, key, value)
+  }
+
+  @tailrec
+  final def parseGlobalMap(
+      remainingBytes: ByteVector,
+      accum: Seq[GlobalPSBTRecord],
+      tx: Transaction): (ByteVector, Seq[GlobalPSBTRecord], Transaction) = {
+    if (remainingBytes.head == 0x00.byteValue) // terminator
+      (remainingBytes.tail, accum, tx)
+
+    val (totalSize, key, value) = parseKeyAndValue(remainingBytes)
+    val rest = remainingBytes.drop(totalSize) //Drop the key-value map
+    require(
+      rest.head == 0x00.byteValue,
+      s"Key-value maps must be separated by the separator char 0x00 was: ${ByteVector(rest.head)}")
+    val next = rest.tail
+    PSBTGlobalKey.fromByte(key.head) match {
+      case UnsignedTransactionKey =>
+        println("DEBUGGING HERE")
+        println(key)
+        println(value)
+        println(next)
+        if (tx != null)
```

Comment:
if we use an `Option[]` like i suggested else where, we can just do a simple pattern match here: 

```scala
txOpt match { 
  case Some(tx) => //do whatever
  case None => throw IllegalArugementException(...)
}
```